core/systemcontractupgrade/system-contract-upgrade.go

```diff
@@ -0,0 +1,116 @@
+			},
+		},
+	}
+}
+
+func UpgradeBuildInSystemContract(config *params.ChainConfig, blockNumber *big.Int, statedb *state.StateDB) {
+	if config == nil || blockNumber == nil || statedb == nil {
+		return
+	}
+	var network string
+	switch core.GenesisHash {
+	/* Add mainnet genesis hash */
+	case params.ChapelGenesisHash:
+		network = chapel
+	case params.RialtoGenesisHash:
```

Comment:
if no match with any network, just return

Reply:
In the further development, we might need to start a local network. This local network might also need system upgrade, so I added a default network type.